src/sentry/users/api/endpoints/user_emails.py

```diff
@@ -38,6 +41,28 @@ class EmailValidator(serializers.Serializer[UserEmail]):
     email = AllowedEmailField(required=True, help_text="The email address to add/remove.")
 
 
+def add_email_signed(email: str, user: User) -> None:
+    """New path for adding email - uses signed URLs"""
+
+    EMAIL_CONFIRMATION_SALT = options.get("user-settings.signed-url-confirmation-emails-salt")
+
+    if email is None:
+        raise InvalidEmailError
+
+    if UserEmail.objects.filter(user=user, email__iexact=email).exists():
+        raise DuplicateEmailError
+
+    # Generate signed data for verification URL
+    signed_data = sign(
```

Comment:
should the verification link expire after some time?

Reply:
Ah, yes, you're correct. The expiration time is checked when you `unsign` the URL. Right now we're using the default, which is 2 days. This aligns with the current behavior of the verification emails, but I should make it explicit. 

https://github.com/getsentry/sentry/blob/4a5544062e8e7e9af004475fb52fc8bd41adea68/src/sentry/utils/signing.py#L32